src/lib/storage/lz4_segment/lz4_segment_iterable.hpp

```diff
@@ -77,35 +90,28 @@ class LZ4SegmentIterable : public PointAccessibleSegmentIterable<LZ4SegmentItera
 
    public:
     // Begin and End Iterator
-    explicit Iterator(ValueIterator data_it, const NullValueIterator null_value_it)
-        : _chunk_offset{0u}, _data_it{data_it}, _null_value_it{null_value_it} {}
+    explicit Iterator(ValueIterator data_it, const std::optional<NullValueIterator> null_value_it)
+        : _chunk_offset{0u}, _data_it{std::move(data_it)}, _null_value_it{std::move(null_value_it)} {}
```

Comment:
Surprised that clang-tidy doesn't complain about this: You cannot move from a const.

Reply:
remove the const?

Reply:
yes